api/src/main/java/org/openmrs/validator/PersonValidator.java

```diff
@@ -121,6 +122,19 @@ private void validateBirthDate(Errors errors, Date birthDate) {
 		rejectIfFutureDate(errors, birthDate, "birthdate");
 		rejectDateIfBefore120YearsAgo(errors, birthDate, "birthdate");
 	}
+
+	/**
+	 * Checks if the death date is in accordance with dead status.
+	 *
+	 * @param errors Stores information about errors encountered during validation
+	 * @param deathDate the deathdate to validate
+	 * @param dead is the dead status
+	 */
+	private void validateDeathDateDead(Errors errors, Date deathDate, boolean dead, boolean deathDateEstimated) {
+		if(deathDate != null && dead == false && deathDateEstimated == false) {
```

Comment:
the above statement checks if deathDate has some non-null value along with dead being set as false ie (the person is not dead but deathDate has been set) which is what we shouldn't allow according to the ticket. The third condition was added explicitly because of one test case in the PersonServeiceTest class. In that test, unlike other tests (where i manually set dead to true whenever deathdate was being set), setting dead to true was leading to a null pointer exception. so i added it. bcoz earlier what i tried was
private void validateDeathDateDead(Errors errors, Date deathDate, boolean dead, boolean deathDateEstimated) {
      if(dead == true || deathDate == null ) {
               return;
       }
       else if(dead == false && deathDate != null) {

       }